hs_core/tests/api/native/test_update_metadata.py

```diff
@@ -21,6 +21,173 @@ def setUp(self):
+        self.assertEqual(cov_point.value['north'], 11.6789)
+        self.assertEqual(cov_point.value['units'], 'radians')
+        del metadata[:]
+
+        # test identifier elements got updated
+        metadata.append({'identifier': {'name': 'newsomeIdentifier', 'url': "http://some.org/004"}})
+        metadata.append({'identifier': {'name': 'newhydroShareIdentifier', 'url': "http://hydroshare.org/003"}})
+        self.res.metadata.update(metadata, self.user)
+
+        # there should be now 4 identifier elements
+        self.assertEqual(self.res.metadata.identifiers.all().count(), 4,
+                         msg="Number of identifier elements not equal to 4.")
+        some_identifier = self.res.metadata.identifiers.filter(name='newsomeIdentifier').first()
+        self.assertEqual(some_identifier.url, "http://some.org/004")
+        hs_identifier = self.res.metadata.identifiers.filter(name='newhydroShareIdentifier').first()
+        self.assertEqual(hs_identifier.url, "http://hydroshare.org/003")
+        del metadata[:]
+
+        # test relation elements got updated
+        metadata.append({'relation': {'type': 'hasPart', 'value': 'http://hydroshare.org/resource/002'}})
+        self.res.metadata.update(metadata, self.user)
+
+        self.assertEqual(self.res.metadata.relations.all().count(), 1,
+                         msg="Number of source elements is not equal to 1")
+        relation = self.res.metadata.relations.filter(type='hasPart').first()
+        self.assertEqual(relation.value, 'http://hydroshare.org/resource/002')
+        del metadata[:]
+
+        # test rights got updated
+        metadata.append({'rights': {'statement': 'This is the new rights statement for this resource', 'url': 'http://rights.ord/002'}})
+        self.res.metadata.update(metadata, self.user)
+        
+        self.assertEqual(self.res.metadata.rights.statement, 'This is the new rights statement for this resource',
+                         msg="Statement of rights did not match.")
+        self.assertEqual(self.res.metadata.rights.url, 'http://rights.ord/002', msg="URL of rights did not match.")
+        del metadata[:]
+
+        # test subject got updated
+        metadata.append({'subject': {'value': 'sub-4'}})
+        metadata.append({'subject': {'value': 'sub-3'}})
+        self.res.metadata.update(metadata, self.user)
+        # there should be 2 subject elements for this resource
+        self.assertEqual(self.res.metadata.subjects.all().count(), 2, msg="Number of subject elements found not be 2.")
+        self.assertIn('sub-3', [sub.value for sub in self.res.metadata.subjects.all()],
+                      msg="Subject element with value of %s does not exist." % 'sub-3')
+        self.assertIn('sub-4', [sub.value for sub in self.res.metadata.subjects.all()],
+                      msg="Subject element with value of %s does not exist." % 'sub-4')
+
     def test_update_science_metadata(self):
```

Comment:
Do we still need this test?

Reply:
good point, this is duplicative with the test added in this PR. I guess that I had thought we might find an issue testing each metadata type with an individual update (which in hindsight was silly). Anyway, I'm just going to close this PR. Thanks Pabitra